module-4/backend/week_one.md

```diff
@@ -5,17 +5,44 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+  The most useful thing I learned from completing the intermissionw week work was basic Javascript fundamentals.
+
 2. What are some tools to help debug JavaScript code?
+  Debugger and console.log(error)
+
 3. What are some tools you need in order to unit test your JavaScript?
+  Mocha
+
 4. What is the syntax for invoking a function?
+  functionName()
+  
 5. What's the difference between `==` and `===` in JavaScript?
+  == shows that the two have the same value
+  === shows that the two have the same value and same Type
+
 6. What's the difference between asynchronous and synchronous JavaScript? 
+  Synchronous JS waits for each operation to complete to executes the next operation.
+  Asynchronous JS executes all operations in the first GO only. The result of each operation will be handled once the result is available.
+
 7. What's a callback function and what are some reasons when we use/need callback functions?
+  A callback is a function to be executed after another function is executed. For example, if you have a function a running then function b, function a could take longer to load which would make function b run first. You can use a callback to ensure function a runs first before function b is executed.
+
 8. What's the biggest difference between a promise and a callback?
+  A promise is either pending, fulfilled, or rejected. After being created if fulfilled, it will do a .then(on-fullfillment) and return the promise. If rejected, it will go to a .then/.catch and then return.
+
 9. How do we setup a route when creating an API with Node and Express?
+  N/A
```

Comment:
this is an old question. next time, be sure you pull the most recent version of master before answering the questions 👍